spec/javascripts/models/beta/story_spec.js

```diff
@@ -246,107 +246,179 @@ describe('Story model', function () {
+  
+              expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
+      });
 
-        expect(changedStory).toEqual({
-          _editing: {
-            labels: [],
-            storyType: type,
-            estimate: '',
-            state: 'unstarted',
-            _isDirty: true
-          }
+      describe(`when estimate is ""`, () => {
+        estimatedValues.forEach(estimatedValue => {
+          describe(`when new estimate is ${estimatedValue}`, () => {
+            it(`change story state to unstarted`, () => {
+              const story = { _editing: { storyType: FEATURE, estimate: '', state: UNSCHEDULED } };
+              const newAttributes = { estimate: estimatedValue };
+  
+              const changedStory = Story.editStory(story, newAttributes);
+      
+              expect(changedStory._editing.state).toEqual(UNSTARTED);
+              expect(changedStory._editing.estimate).toEqual(estimatedValue);
+            });
+          });
         });
       });
-    })
+    });
 
-    it('change story estimate and sets _isDirty to true', () => {
-      const story = { _editing: { estimate: 1, storyType: 'feature', state: 'unstarted' } };
-      const newAttributes = { estimate: 2 };
+    notFeatureTypes.forEach(noFeatureType => {
+      describe(`when story type is ${noFeatureType}`, () => {
+        const otherNotFeatureTypes = notFeatureTypes.filter(type => type !== noFeatureType);
 
-      const changedStory = Story.editStory(story, newAttributes);
+        otherNotFeatureTypes.forEach(otherNotFeatureType => {
+          describe(`when new story type is ${otherNotFeatureType}`, () => {
+            it("change estime to ''", () => {
+              const story = { _editing: { storyType: noFeatureType, estimate: '' } };
+              const newAttributes = { storyType: otherNotFeatureType }
 
-      expect(changedStory).toEqual({
-        _editing: {
-          labels: [],
-          storyType: 'feature',
-          estimate: newAttributes.estimate,
-          state: 'unstarted',
-          _isDirty: true
-        }
-      });
-    });
+              const changedStory = Story.editStory(story, newAttributes);
 
-    const estimateValues = [1, 2, 3, 5, 8];
+              expect(changedStory._editing.storyType).toEqual(otherNotFeatureType);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
 
-    estimateValues.forEach(estimate => {
-      it(`change story state to unstarted when estimate is ${estimate} and sets _isDirty to true`, () => {
-        const story = { _editing: { storyType: 'feature', estimate: '', state: 'unscheduled' } };
-        const newAttributes = { estimate, };
+        Object.keys(status).forEach(item => {
+          describe(`when new status is ${status[item]}`, () => {
```

Comment:
You can apply this style https://github.com/Codeminer42/cm42-central/pull/576/files#diff-e7cad8ffcafac32387e195a4957fe897R257-R273 here.